cluster-autoscaler/core/scale_down.go

```diff
@@ -163,13 +237,21 @@ func copyScaleDownResourcesLimits(source scaleDownResourcesLimits) scaleDownReso
 	return copy
 }
 
-func computeScaleDownResourcesDelta(node *apiv1.Node) (scaleDownResourcesDelta, errors.AutoscalerError) {
+func computeScaleDownResourcesDelta(node *apiv1.Node, nodeGroup cloudprovider.NodeGroup, resourcesWithLimits []string) (scaleDownResourcesDelta, errors.AutoscalerError) {
```

Comment:
You only create resourcesWithLimits, so you can pass it here. I think passing either ResourceLimiter or a bool would be cleaner than this sort of half-calculated value.

Reply:
I guess a bool would be more efficient, given that we call this in a loop. I suspect it doesn't matter too much though.

Reply:
> I guess a bool would be more efficient, given that we call this in a loop. I suspect it doesn't matter too much though

What do you refer to here?

Reply:
> You only create resourcesWithLimits, so you can pass it here. I think passing either ResourceLimiter or a bool would be cleaner than this sort of half-calculated value

I used to have it that way. But I changed it to pass an array here because of usage of method in `getEmptyNodes`. I do not have `ResourceLimiter` there.

Reply:
Right, I missed that. By bool I meant having passing something like `applyGpuLimits bool`. I think it's slightly more readable, but honestly it doesn't really matter. So I guess I'm fine with both.

Reply:
>  So I guess I'm fine with both.

I will leave it then :)